pallets/runtime/tests/src/basic_sto_test.rs

```diff
@@ -0,0 +1,142 @@
+use super::{
+    storage::{register_keyring_account, TestStorage},
+    ExtBuilder,
+};
+use pallet_asset::{self as asset, AssetType};
+use pallet_basic_sto::{self as sto, Fundraiser};
+use pallet_compliance_manager as compliance_manager;
+use pallet_settlement::{self as settlement, VenueDetails};
+use polymesh_primitives::Ticker;
+
+use frame_support::assert_ok;
+use sp_std::convert::TryFrom;
+use test_client::AccountKeyring;
+
+type Origin = <TestStorage as frame_system::Trait>::Origin;
+type Asset = asset::Module<TestStorage>;
+type STO = sto::Module<TestStorage>;
+type ComplianceManager = compliance_manager::Module<TestStorage>;
+type Settlement = settlement::Module<TestStorage>;
+
+#[test]
+fn basic_raise() {
+    ExtBuilder::default()
+        .build()
+        .execute_with(|| basic_raise_with_ext);
+}
+
+fn basic_raise_with_ext() {
+    let alice_did = register_keyring_account(AccountKeyring::Alice).unwrap();
+    let alice_signed = Origin::signed(AccountKeyring::Alice.public());
+    let bob_did = register_keyring_account(AccountKeyring::Bob).unwrap();
+    let bob_signed = Origin::signed(AccountKeyring::Bob.public());
+
+    // Register tokens
+    let ticker = Ticker::try_from(&[0x01][..]).unwrap();
+    let raise_ticker = Ticker::try_from(&[0x02][..]).unwrap();
+    assert_ok!(Asset::create_asset(
+        alice_signed.clone(),
+        vec![0x01].into(),
+        ticker,
+        1_000_000, // Total supply over the limit
+        true,
+        AssetType::default(),
+        vec![],
+        None,
+        Some(alice_did),
+    ));
+    assert_ok!(Asset::create_asset(
```

Comment:
Duplicated modulo signer/did => closure?

Reply:
Ticker name is also different. Since we only use this twice, I can live with some duplications for now.